client/my-sites/email/email-providers-comparison/price/price-information.tsx

```diff
@@ -109,27 +107,48 @@ const FreeTrialPriceInformation = ( {
 
 const PriceInformation = ( {
 	domain,
+	isDomainInCart = false,
 	product,
 }: {
 	domain?: ResponseDomain;
+	isDomainInCart?: boolean;
 	product: ProductListItem | null;
 } ): ReactElement | null => {
+	const { isEligibleForIntroductoryOffer, isEligibleForIntroductoryOfferFreeTrial } =
+		useGetDomainIntroductoryOfferEligibilities( {
+			domain,
+			isDomainInCart,
+			product,
+		} );
+
+	const translate = useTranslate();
+
 	if ( ! product ) {
 		return null;
 	}
 
-	if ( isGoogleWorkspace( product ) && hasDiscount( product ) ) {
+	const isGoogleWorkspaceProduct = isGoogleWorkspace( product );
+
+	if ( ! isGoogleWorkspaceProduct && ! isTitanMail( product ) ) {
+		return null;
+	}
+
+	if ( isGoogleWorkspaceProduct && hasDiscount( product ) && ! isEligibleForIntroductoryOffer ) {
```

Comment:
I see the opportunity to make `PriceInformation` a generic component that does not rely on checking for the email provider type, and rely on the specific price information components handle the checks (e.g. like what's already been done for https://github.com/Automattic/wp-calypso/blob/4328ae9adce0844e3817d18dab00c5b52e25cb55/client/my-sites/email/email-providers-comparison/price/price-information.tsx#L85

In case of discounts, as of now, `hasDiscount( product )` will never return `true` for Titan. And when we do get a chance to test out introductory discount for Titan, it would be handy to have this condition work for both Google and Titan.

Reply:
Done. Hopefully I have interpreted your comments well. I initially wanted to reduce the coverage of this to GW case.